src/m3/ecore/importer.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    Concept,
+    Feature,
+    FeaturesContainer,
+    Metamodel,
+    MetamodelElement,
+    PrimitiveType
+} from "../types.ts"
+import {MetamodelFactory} from "../factory.ts"
+import {hashingIdGen} from "../../id-generation.ts"
+import {
+    asArray,
+    EClassifier,
+    EcoreXml,
+    EStructuralFeature
+} from "./types.ts"
+import {ConceptType} from "../functions.ts"
+
+
+const refPrefix = "#//"
+const deref = (typeDescriptor: string): string =>
+    typeDescriptor.startsWith(refPrefix)
+        ? typeDescriptor.substring(refPrefix.length)
+        : typeDescriptor
+
+export const asLIonCoreMetamodel = (ecoreXml: EcoreXml): Metamodel => {
+    const ePackage = ecoreXml["ecore:EPackage"]
+    // TODO  an Ecore XML can contain multiple EPackage-s
+    const factory = new MetamodelFactory(ePackage["@name"], hashingIdGen({ checkForUniqueHash: true }))
+
+    // TODO  obtain from a built-in, imported M3 instance
+    const stringDatatype = factory.primitiveType("String")
```

Comment:
Yes, we should somehow share the built-in metamodel among the different implementations